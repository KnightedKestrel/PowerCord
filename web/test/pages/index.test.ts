import { mountSuspended } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import Index from '../../app/pages/index.vue';

describe('Index', () => {
    it('can mount the component', async () => {
        const component = await mountSuspended(Index);
        expect(component.vm).toBeTruthy();
    });

    it('matches the snapshot', async () => {
        const component = await mountSuspended(Index);
        expect(component.html()).toMatchSnapshot();
    });
});
