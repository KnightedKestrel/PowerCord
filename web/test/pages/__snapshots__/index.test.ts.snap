// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index > matches the snapshot 1`] = `
"<div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
  <div class="my-12 md:my-20">
    <div class="my-4 flex flex-wrap-reverse items-center justify-center gap-2 md:my-12 md:flex-nowrap md:justify-between">
      <div class="mb-16 flex flex-col gap-6 text-center md:text-left">
        <h1 class="text-4xl"> Powerlifting competition results in <img class="my-1 h-12" src="/assets/img/discord-logo.svg" alt="Discord"></h1>
        <p class="text-xl"> View powerlifting meet data, lifter profiles, and more! </p>
        <div class="flex flex-wrap justify-center gap-4 md:justify-normal"><a href="https://discord.com/oauth2/authorize?client_id=1306740469484486697&amp;permissions=0&amp;scope=bot%20applications.commands" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 gap-2 text-inverted bg-primary hover:bg-primary/75 active:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary btn text-nowrap text-lg"><span class="iconify i-line-md:discord-twotone shrink-0 size-6" aria-hidden="true"></span>Add to Discord
            <!--v-if-->
          </a><a href="https://github.com/KnightedKestrel/PowerCord" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 gap-2 ring ring-inset ring-primary/50 text-primary hover:bg-primary/10 active:bg-primary/10 disabled:bg-transparent aria-disabled:bg-transparent dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus:outline-none focus-visible:ring-2 focus-visible:ring-primary btn text-nowrap text-lg"><span class="iconify i-line-md:github-twotone shrink-0 size-6" aria-hidden="true"></span>View on GitHub
            <!--v-if-->
          </a></div><span><a href="https://www.openpowerlifting.org/" rel="noopener noreferrer" target="_blank" class="focus-visible:outline-primary hover:text-default transition-colors text-primary">Data provided by OpenPowerlifting <span class="iconify i-fe:link-external size-5" aria-hidden="true"></span></a></span>
        <div style="color-scheme: none;"><iframe src="https://status.betterstack.com/badge?theme=dark" width="250" height="30" frameborder="0" scrolling="no"></iframe></div>
      </div>
      <div class="md:max-h-120 relative max-h-80 overflow-y-hidden"><img class="max-h-240 h-full w-full object-cover" src="/assets/img/preview-showcase.png" alt="Screenshot preview showcase for couple commands in Discord">
        <div class="absolute bottom-0 left-0 right-0 h-24 bg-gradient-to-t from-gray-900 to-transparent"></div>
      </div>
    </div>
    <div class="flex flex-wrap gap-4 md:flex-nowrap">
      <div class="basis-sm flex-1">
        <div class="ring-default flex h-full flex-col gap-y-2 rounded-xl bg-gray-800 px-4 py-5 shadow-xl ring"><span class="iconify i-line-md:speed-twotone text-primary text-3xl" aria-hidden="true"></span>
          <h2 class="text-highlighted text-xl font-semibold">Zero Configuration</h2>
          <p class="text-muted">Simply invite the bot to your server and start using commands in Discord right away.</p>
        </div>
      </div>
      <div class="basis-sm flex-1">
        <div class="ring-default flex h-full flex-col gap-y-2 rounded-xl bg-gray-800 px-4 py-5 shadow-xl ring"><span class="iconify i-line-md:download-twotone text-primary text-3xl" aria-hidden="true"></span>
          <h2 class="text-highlighted text-xl font-semibold">Always Up-to-date</h2>
          <p class="text-muted">Built to grab all the latest meet results from the OPL Data Service nightly.</p>
        </div>
      </div>
      <div class="basis-sm flex-1">
        <div class="ring-default flex h-full flex-col gap-y-2 rounded-xl bg-gray-800 px-4 py-5 shadow-xl ring"><span class="iconify i-line-md:cloud-alt-tags-twotone text-primary text-3xl" aria-hidden="true"></span>
          <h2 class="text-highlighted text-xl font-semibold">Open Source</h2>
          <p class="text-muted">Completely open source and available on GitHub for self-hosting or contributions.</p>
        </div>
      </div>
      <div class="basis-sm flex-1">
        <div class="ring-default flex h-full flex-col gap-y-2 rounded-xl bg-gray-800 px-4 py-5 shadow-xl ring"><span class="iconify i-line-md:heart-twotone-half-filled text-primary text-3xl" aria-hidden="true"></span>
          <h2 class="text-highlighted text-xl font-semibold">Support Available</h2>
          <p class="text-muted">Join our support server for help on usage or hosting. Feedback also welcome!</p>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
